### PR TITLE
Add FlashIAP documentation

### PR DESCRIPTION
**Pull request by 0xc0170**
This class provides Flash IAP interface.

I based this on the other drivers we have docs here. It provides overview, API link and an example. I placed this .md in the storage/ folder. Could be in drivers?

[Rendered](https://github.com/0xc0170/mbed_OS_API_Docs/blob/75fb2af4275e062e0e9a44de65363ffeb9a2d431/docs/APIs/storage/flashiap.md)

@c1728p9, @AnotherButler - please review

### CODE REVIEW

File: docs/APIs/storage/flashiap.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- Get sector/flash or page sizes
+- Get flash start
+
+There are some requirements or limitations for flash devices based on the operation. Please read the documentation for each operation.
+
+This class is thread-safe.
+
+## API
+
+You can find the full C++ API [here](https://github.com/ARMmbed/mbed-os/blob/mbed-os-5.4/drivers/FlashIAP.h).
```
Comment (0xc0170) [inline docs/APIs/storage/flashiap.md]: this should be HTML site as in the other drivers docs ? I could not locate it though for 5.4

Comment (iriark01) [inline docs/APIs/storage/flashiap.md]: https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.4/api/FlashIAP_8h_source.html

File: docs/APIs/storage/flashiap.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Flash devices have some requirements and limitations based on the operation. Please read the documentation for each operation.
+
+This class is thread-safe.
+
+## API
+
+You can find the full C++ API [here](https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.4/api/FlashIAP_8h_source.html).
+
+## Example
```
Comment (iriark01) [inline docs/APIs/storage/flashiap.md]: Can't we transclude both the API and the example like we do for the other ones?

File: docs/APIs/storage/flashiap.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Flash devices have some requirements and limitations based on the operation. Please read the documentation for each operation.
+
+This class is thread-safe.
+
+## API
+
+View the full C++ API:
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.4/api/classrtos_1_1Thread.html)
```
Comment (0xc0170) [inline docs/APIs/storage/flashiap.md]: This link yet not correct, waiting for the update to get the proper link?

**Comment by 0xc0170** [comment]
Thanks for edits !

**Comment by 0xc0170** [comment]
cc @ashok-rao

File: docs/APIs/storage/flashiap.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Flash IAP provides the following functionality:
+
+- Read from a flash device.
+- Program data to flash device pages.
+- Erase flash device sectors.
+- Get sector/flash or page sizes.
+- Get the start address of the flash device.
+
+Flash devices have some requirements and limitations based on the operation. Please read the documentation for each operation.
```
Comment (ashok-rao) [inline docs/APIs/storage/flashiap.md]: how about a link to the specific docs here? Optional but would be nice to have.

**Pull request merged**